src/StructuredLogger/ObjectModel/Message.cs

```diff
@@ -8,7 +8,7 @@ public class Message : TextNode, IHasRelevance, IHasSourceFile, IHasLineNumber
         /// <summary>
         /// Let's see if we even need timestamp, it's just eating memory for now
         /// </summary>
-        public DateTime Timestamp { get { return DateTime.MinValue; } set { } }
+        public DateTime Timestamp { get; set; }

```

Comment:
This is going to very significantly increase memory consumption. Are we sure we need this? Is there a way around it that doesn't need storage for every message?

Reply:
This is going to be challenging, I need to look around.

Reply:
Maybe make a `Dictionary<Message, DateTime>`?

I tightly optimized the memory consumption of the core data structures so the viewer can open huge binlogs with minimal memory consumption. In large binlogs there can be millions of messages, so not storing extra 8 bytes on each message was a significant win.

But if it's really difficult, we can keep your change.

Reply:
Still looking.